ballerina/tests/from_xml_with_options.bal

```diff
@@ -0,0 +1,454 @@
+
+    xml xmlVal2 = xml `<author value="55">Robert C. Martin</author>`;
+    record {|
+        string value;
+    |} author = check parseAsType(xmlVal2, sOptions2);
+    test:assertEquals(author.value, "Robert C. Martin");
+}
+
+SourceOptions sOptions3 = {allowDataProjection: false};
+SourceOptions sOptions4 = {attributePrefix: "@", allowDataProjection: false};
+SourceOptions sOptions5 = {textFieldName: "value", allowDataProjection: false};
+
+@test:Config
+function testDisableProjectionForParseStringNegative() returns error? {
+    string xmlStr1 = string `
+        <Employee age="26">
+            <name>Kanth</name>
+            <address zip="12345">
+                <city>Colombo</city>
+                <country>Sri Lanka</country>
+            </address>
+            <title>Software Engineer</title>
+        </Employee>
+    `;
+    Employee|error e = parseString(xmlStr1, sOptions3);
+    test:assertTrue(e is error);
+    test:assertEquals((<error>e).message(), "undefined field 'age' in record 'data.xmldata:Employee'");
+
+    Employee|error e2 = parseString(xmlStr1, sOptions4);
+    test:assertTrue(e2 is error);
+    test:assertEquals((<error>e2).message(), "undefined field 'title' in record 'data.xmldata:Employee'");
+
+    string xmlStr2 = string `
+        <Employee age="26" title="SE">
+            <name>Kanth</name>
+            <address zip="12345">
+                <city>Colombo</city>
+                <country>Sri Lanka</country>
+            </address>
+        </Employee>
+    `;
+    Employee|error e3 = parseString(xmlStr2, sOptions4);
+    test:assertTrue(e3 is error);
+    test:assertEquals((<error>e3).message(), "undefined field '@title' in record 'data.xmldata:Employee'");
+
+    string xmlStr3 = string `<author age="55">Robert C. Martin</author>`;
+    record {|
+        int age;
+    |}|error e4 = parseString(xmlStr3, sOptions5);
+    test:assertTrue(e4 is error);
+    test:assertEquals((<error>e4).message(), "undefined field 'value' in record 'data.xmldata:record {| int age; |}'");
+
+    record {|
+        int age;
+    |}|error e5 = parseString(xmlStr3, sOptions3);
+    test:assertTrue(e5 is error);
+    test:assertEquals((<error>e5).message(), "undefined field '#content' in record 'data.xmldata:record {| int age; |}'");
+}
+
+@test:Config
+function testDisableProjectionForParseAsTypeNegative() returns error? {
+    xml xmlVal1 = xml `
+        <Employee age="26">
+            <name>Kanth</name>
+            <address zip="12345">
+                <city>Colombo</city>
+                <country>Sri Lanka</country>
+            </address>
+            <title>Software Engineer</title>
+        </Employee>
+    `;
+    Employee|error e = parseAsType(xmlVal1, sOptions3);
+    test:assertTrue(e is error);
+    test:assertEquals((<error>e).message(), "undefined field 'age' in record 'data.xmldata:Employee'");
```

Comment:
Minor formatting issue.

Reply:
I have updated ballerina formatting issues [08b2841](https://github.com/ballerina-platform/module-ballerina-data.xmldata/pull/6/commits/08b2841e410c6ef66692c14be703dbe157ac8de9) 
 
When calling the formatter, This line remain unchanged. Are you referring space between cast-expr and the identifier `e`.